pandas-stubs/core/tools/numeric.pyi

```diff
@@ -1 +1,64 @@
-def to_numeric(arg, errors: str = ..., downcast=...): ...
+from typing import (
+    Literal,
+    overload,
+)
+
+import numpy as np
+import pandas as pd
+
+from pandas._typing import (
+    IgnoreRaiseCoerce,
+    Scalar,
+    npt,
+)
+
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> int | float: ...
```

Comment:
Maybe return `float` here as it is the supertype, and then people don't have to `cast`

Reply:
Do type checkers care about nested types?  mypy seems to accept int | float as just float.

Reply:
With this simple code:
```python
def f2() -> Union[int, float]: ...

x = f2()
reveal_type(x)
```
`pyright` reveals the type as `int | float`.  mypy is making a change to no longer consolidate unions (based on issues we found with 0.981).

So if we can avoid returning a `Union` type it makes it better for people using the stubs.

Reply:
I have removed the union returns throughout.